### PR TITLE
#184797807 pagination support to getAll endpoints

### PR DESCRIPTION
**Pull request by Patienceineza**
#### What does this PR do?

this Pr is for pagination support to all endpoints

#### Description of Task to be completed?

this task is composed of creating a way for responses to come in pages and limit responses to reduce time taken for request to be completed

#### How should this be manually tested?

-clone the repo and  run`npm run migarte:all`
- for some endpoints  that are for admin you need to login and put bearer in token
- else just use the end points previously used but add page number and limit per page like this `/reviews?page=2&limit=3`
-  you can try all other. end points for get all and add the query for better results

#### Any background context you want to provide?

n/a

#### What are the relevant pivotal tracker/Trello stories?

https://www.pivotaltracker.com/story/show/184797807

#### Screenshots (if appropriate)
Screenshot 2023-04-04 at 19 17 53
Screenshot 2023-04-04 at 19 19 53

### CODE REVIEW

**Review by edwinnambaje — APPROVED** [review]
LGTM

File: tests/stats.test.js [hunk 1]
```diff
@@ -109,6 +111,7 @@ describe('testing the seller statistics', () => {
       .request(app)
       .get('/users/stats/graph/year')
       .end((error, res) => {
+        console.log(res);
```
Comment (nshutijonathan) [inline tests/stats.test.js]: remove console.log

File: tests/stats.test.js [hunk 2]
```diff
@@ -85,6 +86,7 @@ describe('testing the seller statistics', () => {
       .request(app)
       .get(`/users/stats?start=${start}`)
       .end((error, res) => {
+        console.log(res);
```
Comment (nshutijonathan) [inline tests/stats.test.js]: remove console.log

**Review by nshutijonathan — COMMENTED** [review]
remove console.log

**Review by Zeden12 — APPROVED** [review]
LGTM

File: tests/stats.test.js (hunk 2 above)
Comment (Patienceineza) [reply tests/stats.test.js]: ok

Comment (Patienceineza) [reply tests/stats.test.js]: I removed them

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by blackd44 — APPROVED** [review]
LG

**Pull request merged**